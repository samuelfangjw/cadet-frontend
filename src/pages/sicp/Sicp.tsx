import * as React from 'react';
import { useEffect, useState } from 'react';
import { SicpChapter } from 'src/features/sicp/SicpTypes';

import SicpDisplay from './subcomponents/SicpDisplay';

export type SicpProps = DispatchProps & StateProps & OwnProps;
export type DispatchProps = {
  handleChangeChapter: (chapter: SicpChapter) => void;
};
export type StateProps = {
  chapter: SicpChapter;
};
export type OwnProps = {};

const Sicp: React.FC<SicpProps> = props => {
  const { chapter } = props;

  const [data, setData] = useState([]);

  const getData = () => {
    console.log("effect called");
    fetch('/sicp/test.json', {
      headers: {
        'Content-Type': 'application/json',
        Accept: 'application/json'
      }
    })
      .then(response => {
          console.log(response);
          return response.json();
        })
      .then(myJson => {
          console.log(myJson);
          setData(myJson);
        });
  };

  useEffect(() => getData(), []);

  //TODO check for null json files, get correct type
  let text = "";
  data.forEach(x => text = text + x["text"]);
  const content = text + "test" + chapter;

  return (
    <SicpDisplay content={content} />
  );
};

export default Sicp;
