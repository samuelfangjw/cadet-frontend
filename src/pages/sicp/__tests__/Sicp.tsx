import { mount, shallow } from 'enzyme';
import ReactRouter from 'react-router';
import ContentDisplay from 'src/commons/ContentDisplay';

import Sicp from '../Sicp';
import SicpIndexPage from '../subcomponents/SicpIndexPage';

const routeComponentPropsMock = {
  history: {} as any,
  location: {} as any,
  match: {} as any
};

describe('Sicp renders', () => {
  test('correctly', () => {
    jest.spyOn(ReactRouter, 'useParams').mockReturnValue({ section: 'index' });

    const navbar = <Sicp {...routeComponentPropsMock} />;
    const tree = shallow(navbar);
    expect(tree.debug()).toMatchSnapshot();
  });

  test('index section correctly', () => {
    jest.spyOn(ReactRouter, 'useParams').mockReturnValue({ section: 'index' });
    window.HTMLElement.prototype.scrollIntoView = function () {};

    const navbar = <Sicp {...routeComponentPropsMock} />;
    const wrapper = mount(navbar);
    const display = wrapper.find(ContentDisplay);
    expect(display.prop('display')).toEqual(<SicpIndexPage />);
  });
});
