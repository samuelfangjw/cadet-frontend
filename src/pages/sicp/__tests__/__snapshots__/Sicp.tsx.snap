// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sicp renders correctly 1`] = `
"<ContextProvider value={{...}}>
  <div className=\\"Sicp bp3-running-text bp3-text-large bp3-dark\\">
    <div />
    <MathJaxContext version={3} config={{...}}>
      <ContentDisplay fullWidth={false} display={{...}} loadContentDispatch={[Function: loadContentDispatch]} />
    </MathJaxContext>
  </div>
</ContextProvider>"
`;
